api/v1alpha2/project_types.go

```diff
@@ -150,6 +164,18 @@ type ProjectSpec struct {
 	//+kubebuilder:validation:MinItems:=1
 	//+optional
 	TeamAccess []*ProjectTeamAccess `json:"teamAccess,omitempty"`
+	// DeletionPolicy defines the strategy the Kubernetes operator uses when you delete a project, either manually or by a system event.
+	//
+	// You must use one of the following values:
+	// - `retain`:  When the custom resource is deleted, the operator will not delete the associated project.
+	// - `destroy`: Performs a destroy operation to remove the project. The project must be empty.
+	// - `force`: Forcefully and immediately removes all resources in the project. Once this is completed the operator deletes the project.
+	// Default: `retain`.
+	//
+	//+kubebuilder:validation:Enum:=retain;destroy;force
+	//+kubebuilder:default=retain
+	//+optional
+	DeletionPolicyForProject DeletionPolicyForProject `json:"deletionPolicyForProject,omitempty"`
```

Comment:
Since the deletion policy here remains within the project CR, we can(should?) drop the extra wording and simply use `DeletionPolicy`.